File: src/main/java/Scrumone_Else_Broke_It/WebShop/entity/Product.java
package Scrumone_Else_Broke_It.WebShop.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;


@Entity
@Getter
@Setter
public class Product {

    //Attributes
    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column
    private String category;
    @Column
    private String currency;
    @Column
    private String description;
    @Column
    private String name;
    @Column
    private double price;

    @OneToMany(mappedBy = "product", cascade = {CascadeType.ALL})
    private List<TechnicalDetails> technicalDetails;

}

File: webshop-frontend/projects/shared-lib/src/lib/models/product.ts

export interface Product {
  id: number;
  name: string;
  imageId: number;
  price: number;
  currency: string;
  description: string;
  category: string;
  details: Array<TechnicalDetail>;
}

export interface TechnicalDetail{
  productId: number;
  id: number;
  detailTitle: string;
  detailText: string,
}
